tock/tock/static/js/components/notifications.js

```diff
@@ -0,0 +1,50 @@
+(function setup_notifications() {
+  if("Notification" in window) {
+    Notification.requestPermission().then(function(result) {
```

Comment:
It ignores the promise rejection case.  My logic was that there's nothing to do if this fails for some reason.  I guess as a matter of good practice, though, there's no harm in catching the rejection anyway and logging to the console (or whatever the project standard is), since a rejection is exceptional in this case (the promise should always resolve under normal circumstances).